dependencies.xml

```diff
@@ -64,8 +64,13 @@ Note all install methods after "main" take
     <line_profiler optional='True'/>
     <!-- <ete3 optional='True'/> -->
     <pywavelets optional='True'>1.1</pywavelets>
+    <numdifftools source="pip">0.9.39</numdifftools>
     <fmpy optional='True'/>
     <xmlschema source="pip"/>
+    <pyomo optional='True'>6.4</pyomo>
```

Comment:
@wangcj05 @yoshiurr-INL  My thought is to add something like:
```<pynumero optional='True' install_script='scripts/install_pynumero.sh'/>``` and then when optional are requested, just run the install_pynumero.sh.

Reply:
(After discussing with @yoshiurr-INL , we should also add something like source="custom" to the above.)

Reply:
@yoshiurr-INL is adding a new pyomo-extensions which fixes this.

Reply:
I have added "pyomo" install method for future pyomo installation needs. In establish_conda, if the pyomo installation is used, it will check for the identified package. For future reference, if more pyomo command installations are required, establish_conda is the only file that needs to be updated. For example, if "pyomo custommodel" needs to be ran, the following lines need to be added to establish_conda after line 166:

if [[ ${COMMAND} == *"pyomo-custommodel"* ]];
then
      pyomo pyomo-custommodel
fi

In the dependencies just add the following into <main>:

<pyomo-custommodel source="pyomo" skip_check='True' optional='True'/>

Reply:
<pyomo-custommodel source="pyomo" skip_check='True' optional='True'/>

Reply:
In the dependencies just add the following into main:
pyomo-extensions source="pyomo" skip_check='True' optional='True'/

I apologize for strange comments. It seems adding brackets makes the words in between disappear.